src/libs/duplicate.c

```diff
@@ -148,27 +155,26 @@ static void _lib_duplicate_thumb_press_callback(GtkWidget *widget, GdkEventButto
     {
       dt_develop_t *dev = darktable.develop;
       if(!dev) return;
-      dt_dev_zoom_t zoom;
-      int closeup;
-      float zoom_x, zoom_y;
-      closeup = dt_control_get_dev_closeup();
-      float scale = 0;
-
-      zoom = DT_ZOOM_FIT;
-      scale = dt_dev_get_zoom_scale(dev, DT_ZOOM_FIT, 1.0, 0);
-
-      dt_dev_check_zoom_bounds(dev, &zoom_x, &zoom_y, zoom, closeup, NULL, NULL);
-      dt_control_set_dev_zoom_scale(scale);
-      dt_control_set_dev_zoom(zoom);
-      dt_control_set_dev_closeup(closeup);
-      dt_control_set_dev_zoom_x(zoom_x);
-      dt_control_set_dev_zoom_y(zoom_y);
+
       dt_dev_invalidate(dev);
       dt_control_queue_redraw();
 
       dt_dev_invalidate(darktable.develop);
 
       d->imgid = imgid;
+      int fw, fh;
+      fw = fh = 0;
+      dt_image_get_final_size(imgid, &fw, &fh);
+      if(d->cur_final_width - fw < DUPLICATE_COMPARE_SIZE && d->cur_final_width - fw > -DUPLICATE_COMPARE_SIZE
```

Comment:
this could be replaced by:

```
d->allow_size = (d->cur_final_width - fw <  ... 
                          ...
                          > -DUPLICATE_COMPARE_SIZE);
```